maxim/tests/test_retry_for_maxim_api.py

```
@@ -0,0 +1,128 @@
+        # Assert that the result is True indicating successful upload after retry
+        assert result is True, f"Expected successful upload (True), got {result}"
+
+        # Assert that mock_put was called with expected parameters
+        expected_url = "https://example.com/upload"
+        expected_data = b"test data"
+        expected_headers = {"Content-Type": "text/plain"}
+        expected_timeout = 30
+
+        # Check first call (that failed)
+        first_call = mock_put.call_args_list[0]
+        assert first_call[0][0] == expected_url, f"First call URL mismatch"
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)